rosidl_typesupport_opensplice_cpp/include/rosidl_typesupport_opensplice_cpp/requester.hpp

```diff
@@ -137,10 +139,8 @@ class Requester
       estr = "DomainParticipant::create_topic: failed for response";
       goto fail;
     }
-
-    // Let OpenSplice do any needed encoding
     content_filtered_response_topic_ = participant_->create_contentfilteredtopic(
-      service_name_.c_str(), response_topic_,
+      content_filtered_topic_name.c_str(), response_topic_,
```

Comment:
Connext uses a `ContentFilteredTopic` with the same topic name and filtered on the content too, otherwise the query wouldn't be necessary and we could just use a regular `Topic` object. I haven't had a deeper look at it, but it appears the bug must be somewhere else.